zipkin/src/main/java/zipkin/internal/CorrectForClockSkew.java

```diff
@@ -70,6 +70,18 @@ static void adjust(Node<Span> node, @Nullable ClockSkew skewFromParent) {
     if (skew != null) {
       // the current span's skew may be a different endpoint than skewFromParent, adjust again.
       node.value(adjustTimestamps(node.value(), skew));
+    } else {
+      if (skewFromParent != null) {
+        //Propagate skewFromParent to local spans
+        for (int i = 0, length = node.value().binaryAnnotations.size(); i < length; i++) {
+          BinaryAnnotation b = node.value().binaryAnnotations.get(i);
+          if (b.endpoint == null) continue;
+          if (b.key.equals(Constants.LOCAL_COMPONENT) && ipsMatch(skewFromParent.endpoint, b.endpoint)) {
```

Comment:
just an fyi, but this could result in further maintenance as local spans are not required to have the tag "lc". The local component tag is primarily a tool to use when you don't have any other annotation or binary annotations (as it allows you to have some endpoint attached that you can search with.

So, while this will work for many single-host spans, it won't for all of them. It might be best to change the heuristic or note in a TODO that this won't work for single-host spans that don't contain "lc"